### PR TITLE
Improve MathML examples

### PR DESCRIPTION
**Pull request by pepelsbey**
### Description

- Adds a short intro explaining the example.
- Uses Playground and shows the code.
- Highlights the steps with comments.
- Uses `font-size: smaller` for inline annotations (it nicely aligns with the `\small` key in TeX).
- Adds TeX annotations.

### Motivation

To make examples more useful and allow playing with them.

### CODE REVIEW

File: files/en-us/web/mathml/examples/deriving_the_quadratic_formula/index.md [hunk 1]
```diff
@@ -101,10 +113,13 @@ We take a quadratic equation in its general form, and solve for x:
         </mtd>
         <mtd>
           <mrow>
-            <mtext style="color: red; font-size: 10pt;">Divide out leading coefficient.</mtext>
+            <mtext style="color: red; font-size: smaller">
+              Divide out leading coefficient.
```
Comment (fred-wang) [inline files/en-us/web/mathml/examples/deriving_the_quadratic_formula/index.md:117]: Note that MathML Core does not say leading/trailing spaces are trimmed. Chromium would likely add some space here and below.

File: files/en-us/web/mathml/examples/deriving_the_quadratic_formula/index.md [hunk 2]
```diff
@@ -449,12 +475,26 @@ We take a quadratic equation in its general form, and solve for x:
...
+            <mtext style="color: red; font-size: smaller"></mtext>
           </mrow>
         </mtd>
       </mtr>
     </mtable>
-    <annotation encoding="TeX">TODO</annotation>
+    <!-- Representation in TeX format -->
+    <annotation encoding="application/x-tex">
+      \begin{aligned}
+      ax^2 + bx + c &= 0 \\
```
Comment (fred-wang) [inline files/en-us/web/mathml/examples/deriving_the_quadratic_formula/index.md:486]: I think this should use `&amp;`, right?

File: files/en-us/web/mathml/examples/mathml_pythagorean_theorem/index.md [hunk 3]
```diff
@@ -6,14 +6,120 @@ page-type: guide
...
+            <mi>c</mi>
+            <mn>2</mn>
+          </msup>
+        </mtd>
+      </mtr>
+    </mtable>
+    <!-- Representation in TeX format -->
+    <annotation encoding="application/x-tex">
+      \begin{aligned}
+      (a + b)^2 &= c^2 + 4 \cdot \left( \frac{1}{2} ab \right) \\
```
Comment (fred-wang) [inline files/en-us/web/mathml/examples/mathml_pythagorean_theorem/index.md:117]: Ditto.

File: files/en-us/web/mathml/examples/deriving_the_quadratic_formula/index.md (hunk 1 above)
Comment (pepelsbey) [reply files/en-us/web/mathml/examples/deriving_the_quadratic_formula/index.md]: Good point! Fixed.

File: files/en-us/web/mathml/examples/mathml_pythagorean_theorem/index.md (hunk 3 above)
Comment (fred-wang) [reply files/en-us/web/mathml/examples/mathml_pythagorean_theorem/index.md:117]: OK, `&` immediately followed by something other than `#` or an alphanum seems ok if I read the parsing algo correctly:
  https://html.spec.whatwg.org/multipage/parsing.html#character-reference-state
  https://html.spec.whatwg.org/multipage/parsing.html#flush-code-points-consumed-as-a-character-reference

and indeed the validator does not complain.

The non-normative section https://html.spec.whatwg.org/#syntax-errors mentions it as non-conforming in "Errors involving fragile syntax constructs".  The path causing error would be:
https://html.spec.whatwg.org/multipage/parsing.html#parse-error-unknown-named-character-reference
https://html.spec.whatwg.org/multipage/parsing.html#named-character-reference-state
https://html.spec.whatwg.org/multipage/parsing.html#ambiguous-ampersand-state

**Review by fred-wang — APPROVED** [review]

File: files/en-us/web/mathml/examples/mathml_pythagorean_theorem/index.md (hunk 3 above)
Comment (pepelsbey) [reply files/en-us/web/mathml/examples/mathml_pythagorean_theorem/index.md:117]: There are only two cases in annotations: `&` and `&=`, surrounded by spaces. So it should be fine then.

**Pull request merged**